test/sharness/t0063-daemon-init.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+#
+# Copyright (c) 2014 Juan Batiz-Benet
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="Test daemon --init command"
+
+. lib/test-lib.sh
+
+# We don't want the normal test_init_ipfs but we need to make sure the
+# IPFS_PATH is set correctly.
+export IPFS_PATH="$(pwd)/.ipfs"
+
+# safety check since we will be removing the directory
+if [ -e "$IPFS_PATH" ]; then
+	echo "$IPFS_PATH exists"
+	exit 1
+fi
+
+test_ipfs_daemon_init() {
+	# Doing it manually since we want to launch the daemon with an
+	# empty or non-existent repo; the normal
+	# test_launch_ipfs_daemon does not work since it assumes the
+	# repo was created a particular way with regard to the API
+	# server.
+
+	test_expect_success "'ipfs daemon --init' succeeds" '
+		ipfs daemon --init >actual_daemon 2>daemon_err &
```

Comment:
Actually because I use a simple sleep and the init takes some time the test will pass as the daemon won't try to start right away.  I tested the tests with out my fix and the test fails because the daemon fails right away if there is a problem reading the repo.

If I up the timeout the daemon error output will make it fairly clear that there is a port conflict.
```
Error: serveHTTPApi: manet.Listen(/ip4/127.0.0.1/tcp/5001) failed: listen tcp4 127.0.0.1:5001: bind: address already in use
```
but you need to run the test with `-v`.

I am unsure how to use `netcat` to test for port conflict.  `netcat -w0 localhost 8080` does not work on my machine, I get "invalid wait-time 0".  Port testing doesn't seam like a documented use of `netcat` (at least the version (v1.10-41) I have installed on my machine) 

The sleep is 2 seconds now, I can lower it to 1 if you think it would be better.